experimental/oak_baremetal_kernel/src/lib.rs

```diff
@@ -91,18 +92,12 @@ enum ChannelType {
     SimpleIo,
 }
 
-fn main(protocol: &str, kernel_args: args::Args) -> ! {
+fn main<H: oak_idl::Handler>(protocol: &str, kernel_args: args::Args, idl_service: H) -> ! {
```

Comment:
I'd go one step further refactoring all of this code, but as Conrad said above, it should likely be in a separate PR.

What I imagine is `start_kernel()` returning some kind of data structure -- for now, it might be just `Box<dyn Channel>`, and then in `oak_baremetal_app_crosvm`, instead of calling start_kernel with the idl_service as an argument, you call `oak_baremetal_communication_channel::server::start_blocking_server` in there. No refererences to IDL in the kernel required!

Reply:
Done
Also moved `main` to the `app_crosvm` instead of `kernel`